release-notes/7.0/preview/api-diff/preview1/Microsoft.AspNetCore.App/7.0-preview1_Microsoft.Extensions.FileSystemGlobbing.Internal.PathSegments.md

```diff
@@ -0,0 +1,11 @@
+# Microsoft.Extensions.FileSystemGlobbing.Internal.PathSegments
+
+``` diff
+ namespace Microsoft.Extensions.FileSystemGlobbing.Internal.PathSegments {
+     public class LiteralPathSegment : IPathSegment {
+-        public override bool Equals(object obj);
++        public override bool Equals(object? obj);
```

Comment:
@eerhardt, we've typically be putting `[NotNullWhen(true)]` on the `Equals(object? obj)` parameter, assuming it makes sense (which it does in the 99.9% case).

Reply:
It's there in the code:

https://github.com/dotnet/runtime/blob/385fc8832b1d4160abe13645054b65458b6a12f8/src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/Internal/PathSegments/LiteralPathSegment.cs#L29

and in the ref:

https://github.com/dotnet/runtime/blob/385fc8832b1d4160abe13645054b65458b6a12f8/src/libraries/Microsoft.Extensions.FileSystemGlobbing/ref/Microsoft.Extensions.FileSystemGlobbing.cs#L153

@carlossanlop ? Is this a bug in the diff tool?

Reply:
Probably. I would have to investigate. First time I run the tool.

I can confirm that it's also showing up like you mentioned in the preview1 branch (wanted to make sure the annotation wasn't more recent): https://github.com/dotnet/runtime/blob/release/7.0-preview1/src/libraries/Microsoft.Extensions.FileSystemGlobbing/ref/Microsoft.Extensions.FileSystemGlobbing.cs#L153

Reply:
Since this might be an actual limitation on the tool, there's no action for me to take on this PR.

I opened this arcade issue to track the bug: https://github.com/dotnet/arcade/issues/8516